sdk/core/azure-core/test/ut/transport_adapter_curl.cpp

```diff
@@ -36,85 +36,91 @@ namespace Azure { namespace Core { namespace Test {
     }
   } // namespace
 
-  /***********************  Unique Tests for Libcurl   ********************************/
-  TEST_P(TransportAdapter, connectionPoolTest)
-  {
-    Azure::Core::Http::Url host("http://httpbin.org/get");
-    Azure::Core::Http::CurlConnectionPool::ClearIndex();
-
-    auto threadRoutine = [&]() {
-      using namespace std::chrono_literals;
-      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
-      auto response = m_pipeline->Send(Azure::Core::GetApplicationContext(), request);
-      checkResponseCode(response->GetStatusCode());
-      auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
-      CheckBodyFromBuffer(*response, expectedResponseBodySize);
-      std::this_thread::sleep_for(1s);
-    };
-
-    std::thread t1(threadRoutine);
-    std::thread t2(threadRoutine);
-    t1.join();
-    t2.join();
-
-    // 2 connections must be available at this point
-    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 2);
-
-    std::thread t3(threadRoutine);
-    std::thread t4(threadRoutine);
-    std::thread t5(threadRoutine);
-    t3.join();
-    t4.join();
-    t5.join();
-
-    // Two connections re-used plus one connection created
-    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 3);
-
-#ifdef RUN_LONG_UNIT_TESTS
+/***********************  Unique Tests for Libcurl   ********************************/
+//  Disabling test with TEST_DISABLED. The test name cannot be changed because it depends on a
+//  friend class definition. Hence, it can't use the gtest DISABLE_
+#if !defined(TEST_DISABLED)
```

Comment:
This double negative is confusing :p

You don't need the `add_compile_definitions(TEST_DISABLED)` above at all.

Just using `#if TEST_DISABLED` without defining any compiler preprocessor flag is sufficient.

Reply:
well, I find it less confusing that the other way around XD (lol)

just using `#if TEST_DISABLED` means that you are adding disabled code only when  `TEST_DISABLED` is defined. But does the name `TEST_DISABLED` by itself means `do not add disabled tests`.  It's like creating a boolean `isDead` which true means it's alive.

Will you agree with using a name like `INCLUDE_DISABLED_TESTS` better?
Here,

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/1077